wxpython not working on MacOS-ARM64 (M1)

<!--
Thanks for reporting your issue.
Please fill out the sections below.
-->

- [X] I read [the conda-forge documentation](https://conda-forge.org/docs/user/introduction.html#how-can-i-install-packages-from-conda-forge) and could not find the solution for my problem there.

Issue:

I have installed wxpython using miniforge. wxpython apps will not run in pythonw because that is broken (https://github.com/conda-forge/python.app-feedstock/issues/22), but if I write my own applescript to launch python in a framework, I still hit problems running wxdemo. 

When the app starts, I get a couple of pages with errors like this:
```
objc[2371]: Class wxNSAppController is implemented in both /Users/toby/miniforge3/lib/libwx_osx_cocoau_core-3.1-arm64-apple-darwin20.0.0.dylib (0x105bc0540) and /Users/toby/miniforge3/lib/libwx_osx_cocoau_core-3.1.5.0.0.dylib (0x10711c540). One of the two will be used. Which one is undefined.
```
and 
```
/Users/runner/miniforge3/conda-bld/wxpython_1614965070577/work/ext/wxWidgets/src/common/object.cpp(239): assert "classTable->Get(m_className) == __null" failed in Register(): Class "wxCommandEvent" already in RTTI table - have you used wxIMPLEMENT_DYNAMIC_CLASS() multiple times or linked some object file twice)?
```

finally a window is opened, but is largely blank with this Python error: 
```
Traceback (most recent call last):
  File "/Users/toby/miniforge3/lib/python3.9/site-packages/wx/core.py", line 2326, in Notify
    self.notify()
  File "/Users/toby/miniforge3/lib/python3.9/site-packages/wx/core.py", line 3532, in Notify
    self.result = self.callable(*self.args, **self.kwargs)
  File "/private/tmp/wxPython-demo-4.1.1/demo/Main.py", line 2620, in ShowMain
    frame = wxPythonDemo(None, "wxPython: (A Demonstration)")
  File "/private/tmp/wxPython-demo-4.1.1/demo/Main.py", line 1550, in __init__
    self.ovr = wx.html.HtmlWindow(panel, -1, size=(400, 400))
wx._core.wxAssertionError: C++ assertion ""Assert failure"" failed at /Users/runner/miniforge3/conda-bld/wxpython_1614965070577/work/ext/wxWidgets/src/html/htmlpars.cpp(316) in AddTag(): No HTML tag handlers registered, is your program linked correctly (you might need to use FORCE_WXHTML_MODULES)?
``` 

As a note for anyone also trying to get together Python and wxpython etc in a native Apple Silicon version. wxpython from Homebrew seems to run fine. ~~but see workaround here: https://github.com/Homebrew/homebrew-core/issues/79562~~ (now fixed) 

<br/>
Environment (<code>conda list</code>):
<details>

```
(base) toby@bht21 ~ % conda list
# packages in environment at /Users/toby/mf4.10.1:
#
# Name                    Version                   Build  Channel
backports                 1.0                        py_2    conda-forge
backports.lzma            0.0.14           py39h6167482_3    conda-forge
brotlipy                  0.7.0           py39h46acfd9_1001    conda-forge
ca-certificates           2021.5.30            h4653dfc_0    conda-forge
certifi                   2021.5.30        py39h2804cbe_0    conda-forge
cffi                      1.14.5           py39h702c04f_0    conda-forge
chardet                   4.0.0            py39h2804cbe_1    conda-forge
conda                     4.10.1           py39h2804cbe_0    conda-forge
conda-package-handling    1.7.3            py39h5161555_0    conda-forge
cryptography              3.4.7            py39h73257c9_0    conda-forge
expat                     2.4.1                hbdafb3b_0    conda-forge
gettext                   0.19.8.1          hea66d9f_1005    conda-forge
idna                      2.10               pyh9f0ad1d_0    conda-forge
jpeg                      9d                   h27ca646_0    conda-forge
libcxx                    11.1.0               h168391b_0    conda-forge
libffi                    3.3                  h9f76cd9_2    conda-forge
libiconv                  1.16                 h642e427_0    conda-forge
libpng                    1.6.37               hf7e6567_2    conda-forge
ncurses                   6.2                  h9aa5885_4    conda-forge
openssl                   1.1.1k               h27ca646_0    conda-forge
pathlib2                  2.3.5            py39h2804cbe_3    conda-forge
pip                       21.1.2             pyhd8ed1ab_0    conda-forge
pthread-stubs             0.4               h27ca646_1001    conda-forge
pycosat                   0.6.3           py39h46acfd9_1006    conda-forge
pycparser                 2.20               pyh9f0ad1d_2    conda-forge
pyopenssl                 20.0.1             pyhd8ed1ab_0    conda-forge
pypubsub                  4.0.3                      py_0    conda-forge
pysocks                   1.7.1            py39h2804cbe_3    conda-forge
python                    3.9.5           h54d631c_0_cpython    conda-forge
python.app                1.3              py39hf68e5b1_4    conda-forge
python_abi                3.9                      1_cp39    conda-forge
readline                  8.1                  hedafd6a_0    conda-forge
requests                  2.25.1             pyhd3deb0d_0    conda-forge
ruamel_yaml               0.15.80         py39h46acfd9_1004    conda-forge
setuptools                49.6.0           py39h2804cbe_3    conda-forge
six                       1.16.0             pyh6c4a22f_0    conda-forge
sqlite                    3.36.0               h72a2b83_0    conda-forge
tk                        8.6.10               hf7e6567_1    conda-forge
tqdm                      4.61.1             pyhd8ed1ab_0    conda-forge
tzdata                    2021a                he74cb21_0    conda-forge
urllib3                   1.26.5             pyhd8ed1ab_0    conda-forge
wheel                     0.36.2             pyhd3deb0d_0    conda-forge
wxpython                  4.1.1            py39h28cb02c_1    conda-forge
xz                        5.2.5                h642e427_1    conda-forge
yaml                      0.2.5                h642e427_0    conda-forge
zlib                      1.2.11            h31e879b_1009    conda-forge
(base) toby@bht21 ~ % 


```
</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>

```
(base) toby@bht21 ~ % conda info

     active environment : base
    active env location : /Users/toby/mf4.10.1
            shell level : 1
       user config file : /Users/toby/.condarc
 populated config files : /Users/toby/mf4.10.1/.condarc
                          /Users/toby/.condarc
          conda version : 4.10.1
    conda-build version : not installed
         python version : 3.9.5.final.0
       virtual packages : __osx=11.2.3=0
                          __unix=0=0
                          __archspec=1=arm64
       base environment : /Users/toby/mf4.10.1  (writable)
      conda av data dir : /Users/toby/mf4.10.1/etc/conda
  conda av metadata url : https://repo.anaconda.com/pkgs/main
           channel URLs : https://conda.anaconda.org/conda-forge/osx-arm64
                          https://conda.anaconda.org/conda-forge/noarch
          package cache : /Users/toby/mf4.10.1/pkgs
                          /Users/toby/.conda/pkgs
       envs directories : /Users/toby/mf4.10.1/envs
                          /Users/toby/.conda/envs
               platform : osx-arm64
             user-agent : conda/4.10.1 requests/2.25.1 CPython/3.9.5 Darwin/20.3.0 OSX/11.2.3
                UID:GID : 3328:1113
             netrc file : None
           offline mode : False
```
</details>

I have been getting a similar message on my M1 Mac as well. It looks like the problem is that the wx libraries are all copies of one another instead of symbolic links. For example, in `${CONDA_PREFIX}/lib` in an environment with [wxpython=4.1.1](https://conda.anaconda.org/conda-forge/osx-arm64/wxpython-4.1.1-py310h88f142e_5.tar.bz2) there are libraries that look like
```
14320 -rwxrwxr-x    3 bkpoon  staff   7.0M Dec  1 09:31 libwx_osx_cocoau_core-3.1-arm64-apple-darwin20.0.0.dylib                    
14320 -rwxrwxr-x    3 bkpoon  staff   7.0M Dec  1 09:31 libwx_osx_cocoau_core-3.1.5.0.0.dylib                                       
14320 -rwxrwxr-x    3 bkpoon  staff   7.0M Dec  1 09:30 libwx_osx_cocoau_core-3.1.5.dylib
```
The regular `osx-64` package has
```
14704 -rwxr-xr-x@ 1 bkpoon  staff   7.2M Mar 20 08:15 osx-64/lib/libwx_osx_cocoau_core-3.1.5.0.0.dylib
    0 lrwxr-xr-x@ 1 bkpoon  staff    37B Mar 20 08:15 osx-64/lib/libwx_osx_cocoau_core-3.1.5.dylib -> libwx_osx_cocoau_core-3.1.5.0.0.dylib
    0 lrwxr-xr-x@ 1 bkpoon  staff    37B Mar 20 08:15 osx-64/lib/libwx_osx_cocoau_core-3.1.dylib -> libwx_osx_cocoau_core-3.1.5.0.0.dylib
```

You can also check by extracting the wxPython conda packages for the `osx-64` and `osx-arm64` platforms and looking at the `lib` directory.

When I try to import something from `wx`, I get warnings like this
```
objc[26990]: Class wxStaticBitmapView is implemented in both /Users/bkpoon/software/phenix/conda_base/lib/libwx_osx_cocoau_core-3.1-
arm64-apple-darwin20.0.0.dylib (0x102ffd8c8) and /Users/bkpoon/software/phenix/conda_base/lib/libwx_osx_cocoau_core-3.1.5.0.0.dylib 
(0x1266a18c8). One of the two will be used. Which one is undefined. 
```

which causes an exception that looks like
```
wx._core.wxAssertionError: C++ assertion "classTable->Get(m_className) == __null" failed at /Users/runner/miniforge3/conda-bld/wxpython_1638375572653/work/ext/wxWidgets/src/common/object.cpp(239) in Register(): Class "wxSashLayoutWindow" already in RTTI table - have you used wxIMPLEMENT_DYNAMIC_CLASS() multiple times or linked some object file twice)?
```

It looks like the copies cause Python to think there are different objects with the same name and the fix would be to change the copies in `osx-arm64` package to symbolic links like the `osx-64` package. I did a basic test where I manually deleted the copies of one of the libraries and replaced them with symbolic links to one of the files. I was able to get the import to work, but I have not changed all the libraries so I do not expect my GUI to be fully functional.

I can maybe make a PR next week? I would maybe add a section to `build.sh` that only runs on `osx-arm64` builds that picks one of the libraries as the original and replaces the others with symbolic links. I would follow the same linking as the `osx-64` package.

Or could this be a cross-compilation issue?

--- a/.azure-pipelines/azure-pipelines-linux.yml
+++ b/.azure-pipelines/azure-pipelines-linux.yml
@@ -12,18 +12,22 @@ jobs:
         CONFIG: linux_64_python3.10.____cpython
         UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
+        SHORT_CONFIG: linux_64_python3.10.____cpython
       linux_64_python3.7.____cpython:
         CONFIG: linux_64_python3.7.____cpython
         UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
+        SHORT_CONFIG: linux_64_python3.7.____cpython
       linux_64_python3.8.____cpython:
         CONFIG: linux_64_python3.8.____cpython
         UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
+        SHORT_CONFIG: linux_64_python3.8.____cpython
       linux_64_python3.9.____cpython:
         CONFIG: linux_64_python3.9.____cpython
         UPLOAD_PACKAGES: 'True'
         DOCKER_IMAGE: quay.io/condaforge/linux-anvil-cos7-x86_64
+        SHORT_CONFIG: linux_64_python3.9.____cpython
   timeoutInMinutes: 360
 
   steps:
@@ -55,3 +59,32 @@ jobs:
       BINSTAR_TOKEN: $(BINSTAR_TOKEN)
       FEEDSTOCK_TOKEN: $(FEEDSTOCK_TOKEN)
       STAGING_BINSTAR_TOKEN: $(STAGING_BINSTAR_TOKEN)
+  - script: |
+        export CI=azure
+        export CI_RUN_ID=$(build.BuildNumber).$(system.JobAttempt)
+        export FEEDSTOCK_NAME=$(basename ${BUILD_REPOSITORY_NAME})
+        export CONDA_BLD_DIR=build_artifacts
+        export ARTIFACT_STAGING_DIR="$(Build.ArtifactStagingDirectory)"
+        # Archive everything in CONDA_BLD_DIR except environments
+        export BLD_ARTIFACT_PREFIX=conda_artifacts
+        if [[ "$AGENT_JOBSTATUS" == "Failed" ]]; then
+          # Archive the CONDA_BLD_DIR environments only when the job fails
+          export ENV_ARTIFACT_PREFIX=conda_envs
+        fi
+        ./.scripts/create_conda_build_artifacts.sh
+    displayName: Prepare conda build artifacts
+    condition: succeededOrFailed()
+
+  - task: PublishPipelineArtifact@1
+    displayName: Store conda build artifacts
+    condition: not(eq(variables.BLD_ARTIFACT_PATH, ''))
+    inputs:
+      targetPath: $(BLD_ARTIFACT_PATH)
+      artifactName: $(BLD_ARTIFACT_NAME)
+
+  - task: PublishPipelineArtifact@1
+    displayName: Store conda build environment artifacts
+    condition: not(eq(variables.ENV_ARTIFACT_PATH, ''))
+    inputs:
+      targetPath: $(ENV_ARTIFACT_PATH)
+      artifactName: $(ENV_ARTIFACT_NAME)

--- a/.azure-pipelines/azure-pipelines-osx.yml
+++ b/.azure-pipelines/azure-pipelines-osx.yml
@@ -11,24 +11,31 @@ jobs:
       osx_64_python3.10.____cpython:
         CONFIG: osx_64_python3.10.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_64_python3.10.____cpython
       osx_64_python3.7.____cpython:
         CONFIG: osx_64_python3.7.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_64_python3.7.____cpython
       osx_64_python3.8.____cpython:
         CONFIG: osx_64_python3.8.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_64_python3.8.____cpython
       osx_64_python3.9.____cpython:
         CONFIG: osx_64_python3.9.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_64_python3.9.____cpython
       osx_arm64_python3.10.____cpython:
         CONFIG: osx_arm64_python3.10.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_arm64_python3.10.____cpython
       osx_arm64_python3.8.____cpython:
         CONFIG: osx_arm64_python3.8.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_arm64_python3.8.____cpython
       osx_arm64_python3.9.____cpython:
         CONFIG: osx_arm64_python3.9.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: osx_arm64_python3.9.____cpython
   timeoutInMinutes: 360
 
   steps:
@@ -49,3 +56,32 @@ jobs:
       BINSTAR_TOKEN: $(BINSTAR_TOKEN)
       FEEDSTOCK_TOKEN: $(FEEDSTOCK_TOKEN)
       STAGING_BINSTAR_TOKEN: $(STAGING_BINSTAR_TOKEN)
+  - script: |
+      export CI=azure
+      export CI_RUN_ID=$(build.BuildNumber).$(system.JobAttempt)
+      export FEEDSTOCK_NAME=$(basename ${BUILD_REPOSITORY_NAME})
+      export CONDA_BLD_DIR=/Users/runner/miniforge3/conda-bld
+      export ARTIFACT_STAGING_DIR="$(Build.ArtifactStagingDirectory)"
+      # Archive everything in CONDA_BLD_DIR except environments
+      export BLD_ARTIFACT_PREFIX=conda_artifacts
+      if [[ "$AGENT_JOBSTATUS" == "Failed" ]]; then
+        # Archive the CONDA_BLD_DIR environments only when the job fails
+        export ENV_ARTIFACT_PREFIX=conda_envs
+      fi
+      ./.scripts/create_conda_build_artifacts.sh
+    displayName: Prepare conda build artifacts
+    condition: succeededOrFailed()
+
+  - task: PublishPipelineArtifact@1
+    displayName: Store conda build artifacts
+    condition: not(eq(variables.BLD_ARTIFACT_PATH, ''))
+    inputs:
+      targetPath: $(BLD_ARTIFACT_PATH)
+      artifactName: $(BLD_ARTIFACT_NAME)
+
+  - task: PublishPipelineArtifact@1
+    displayName: Store conda build environment artifacts
+    condition: not(eq(variables.ENV_ARTIFACT_PATH, ''))
+    inputs:
+      targetPath: $(ENV_ARTIFACT_PATH)
+      artifactName: $(ENV_ARTIFACT_NAME)

--- a/.azure-pipelines/azure-pipelines-win.yml
+++ b/.azure-pipelines/azure-pipelines-win.yml
@@ -11,15 +11,19 @@ jobs:
       win_64_python3.10.____cpython:
         CONFIG: win_64_python3.10.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: win_64_python3.10.____cpython
       win_64_python3.7.____cpython:
         CONFIG: win_64_python3.7.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: win_64_python3.7.____cpython
       win_64_python3.8.____cpython:
         CONFIG: win_64_python3.8.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: win_64_python3.8.____cpython
       win_64_python3.9.____cpython:
         CONFIG: win_64_python3.9.____cpython
         UPLOAD_PACKAGES: 'True'
+        SHORT_CONFIG: win_64_python3.9.____cpython
   timeoutInMinutes: 360
   variables:
     CONDA_BLD_PATH: D:\\bld\\
@@ -99,6 +103,33 @@ jobs:
       env:
         PYTHONUNBUFFERED: 1
       condition: not(contains(variables['CONFIG'], 'vs2008'))
+    - script: |
+        set CI=azure
+        set CI_RUN_ID=$(build.BuildNumber).$(system.JobAttempt)
+        set FEEDSTOCK_NAME=$(build.Repository.Name)
+        set ARTIFACT_STAGING_DIR=$(Build.ArtifactStagingDirectory)
+        set CONDA_BLD_DIR=$(CONDA_BLD_PATH)
+        set BLD_ARTIFACT_PREFIX=conda_artifacts
+        if "%AGENT_JOBSTATUS%" == "Failed" (
+            set ENV_ARTIFACT_PREFIX=conda_envs
+        )
+        call ".scripts\create_conda_build_artifacts.bat"
+      displayName: Prepare conda build artifacts
+      condition: succeededOrFailed()
+
+    - task: PublishPipelineArtifact@1
+      displayName: Store conda build artifacts
+      condition: not(eq(variables.BLD_ARTIFACT_PATH, ''))
+      inputs:
+        targetPath: $(BLD_ARTIFACT_PATH)
+        artifactName: $(BLD_ARTIFACT_NAME)
+
+    - task: PublishPipelineArtifact@1
+      displayName: Store conda build environment artifacts
+      condition: not(eq(variables.ENV_ARTIFACT_PATH, ''))
+      inputs:
+        targetPath: $(ENV_ARTIFACT_PATH)
+        artifactName: $(ENV_ARTIFACT_NAME)
     - script: |
         set "FEEDSTOCK_NAME=%BUILD_REPOSITORY_NAME:*/=%"
         call activate base

--- a/.scripts/create_conda_build_artifacts.bat
+++ b/.scripts/create_conda_build_artifacts.bat
@@ -1,0 +1,80 @@
+setlocal enableextensions enabledelayedexpansion
+
+rem INPUTS (environment variables that need to be set before calling this script):
+rem
+rem CI (azure/github_actions/UNSET)
+rem CI_RUN_ID (unique identifier for the CI job run)
+rem FEEDSTOCK_NAME
+rem CONFIG (build matrix configuration string)
+rem SHORT_CONFIG (uniquely-shortened configuration string)
+rem CONDA_BLD_DIR (path to the conda-bld directory)
+rem ARTIFACT_STAGING_DIR (use working directory if unset)
+rem BLD_ARTIFACT_PREFIX (prefix for the conda build artifact name, skip if unset)
+rem ENV_ARTIFACT_PREFIX (prefix for the conda build environments artifact name, skip if unset)
+
+rem OUTPUTS
+rem
+rem BLD_ARTIFACT_NAME
+rem BLD_ARTIFACT_PATH
+rem ENV_ARTIFACT_NAME
+rem ENV_ARTIFACT_PATH
+
+rem Check that the conda-build directory exists
+if not exist %CONDA_BLD_DIR% (
+    echo conda-build directory does not exist
+    exit 1
+)
+
+if not defined ARTIFACT_STAGING_DIR (
+    rem Set staging dir to the working dir
+    set ARTIFACT_STAGING_DIR=%cd%
+)
+
+rem Set a unique ID for the artifact(s), specialized for this particular job run
+set ARTIFACT_UNIQUE_ID=%CI_RUN_ID%_%CONFIG%
+if not "%ARTIFACT_UNIQUE_ID%" == "%ARTIFACT_UNIQUE_ID:~0,80%" (
+    set ARTIFACT_UNIQUE_ID=%CI_RUN_ID%_%SHORT_CONFIG%
+)
+
+rem Set a descriptive ID for the archive(s), specialized for this particular job run
+set ARCHIVE_UNIQUE_ID=%CI_RUN_ID%_%CONFIG%
+
+rem Make the build artifact zip
+if defined BLD_ARTIFACT_PREFIX (
+    set BLD_ARTIFACT_NAME=%BLD_ARTIFACT_PREFIX%_%ARTIFACT_UNIQUE_ID%
+    echo BLD_ARTIFACT_NAME: !BLD_ARTIFACT_NAME!
+
+    set "BLD_ARTIFACT_PATH=%ARTIFACT_STAGING_DIR%\%FEEDSTOCK_NAME%_%BLD_ARTIFACT_PREFIX%_%ARCHIVE_UNIQUE_ID%.zip"
+    7z a "!BLD_ARTIFACT_PATH!" "%CONDA_BLD_DIR%" -xr^^!.git/ -xr^^!_*_env*/ -xr^^!*_cache/ -bb
+    if errorlevel 1 exit 1
+    echo BLD_ARTIFACT_PATH: !BLD_ARTIFACT_PATH!
+
+    if "%CI%" == "azure" (
+        echo ##vso[task.setVariable variable=BLD_ARTIFACT_NAME]!BLD_ARTIFACT_NAME!
+        echo ##vso[task.setVariable variable=BLD_ARTIFACT_PATH]!BLD_ARTIFACT_PATH!
+    )
+    if "%CI%" == "github_actions" (
+        echo ::set-output name=BLD_ARTIFACT_NAME::!BLD_ARTIFACT_NAME!
+        echo ::set-output name=BLD_ARTIFACT_PATH::!BLD_ARTIFACT_PATH!
+    )
+)
+
+rem Make the environments artifact zip
+if defined ENV_ARTIFACT_PREFIX (
+    set ENV_ARTIFACT_NAME=!ENV_ARTIFACT_PREFIX!_%ARTIFACT_UNIQUE_ID%
+    echo ENV_ARTIFACT_NAME: !ENV_ARTIFACT_NAME!
+
+    set "ENV_ARTIFACT_PATH=%ARTIFACT_STAGING_DIR%\%FEEDSTOCK_NAME%_%ENV_ARTIFACT_PREFIX%_%ARCHIVE_UNIQUE_ID%.zip"
+    7z a "!ENV_ARTIFACT_PATH!" -r "%CONDA_BLD_DIR%"/_*_env*/ -bb
+    if errorlevel 1 exit 1
+    echo ENV_ARTIFACT_PATH: !ENV_ARTIFACT_PATH!
+
+    if "%CI%" == "azure" (
+        echo ##vso[task.setVariable variable=ENV_ARTIFACT_NAME]!ENV_ARTIFACT_NAME!
+        echo ##vso[task.setVariable variable=ENV_ARTIFACT_PATH]!ENV_ARTIFACT_PATH!
+    )
+    if "%CI%" == "github_actions" (
+        echo ::set-output name=ENV_ARTIFACT_NAME::!ENV_ARTIFACT_NAME!
+        echo ::set-output name=ENV_ARTIFACT_PATH::!ENV_ARTIFACT_PATH!
+    )
+)

--- a/.scripts/create_conda_build_artifacts.sh
+++ b/.scripts/create_conda_build_artifacts.sh
@@ -1,0 +1,113 @@
+#!/usr/bin/env bash
+
+# INPUTS (environment variables that need to be set before calling this script):
+#
+# CI (azure/github_actions/UNSET)
+# CI_RUN_ID (unique identifier for the CI job run)
+# FEEDSTOCK_NAME
+# CONFIG (build matrix configuration string)
+# SHORT_CONFIG (uniquely-shortened configuration string)
+# CONDA_BLD_DIR (path to the conda-bld directory)
+# ARTIFACT_STAGING_DIR (use working directory if unset)
+# BLD_ARTIFACT_PREFIX (prefix for the conda build artifact name, skip if unset)
+# ENV_ARTIFACT_PREFIX (prefix for the conda build environments artifact name, skip if unset)
+
+# OUTPUTS
+#
+# BLD_ARTIFACT_NAME
+# BLD_ARTIFACT_PATH
+# ENV_ARTIFACT_NAME
+# ENV_ARTIFACT_PATH
+
+source .scripts/logging_utils.sh
+
+# DON'T do set -x, because it results in double echo-ing pipeline commands
+# and that might end up inserting extraneous quotation marks in output variables
+set -e
+
+# Check that the conda-build directory exists
+if [ ! -d "$CONDA_BLD_DIR" ]; then
+    echo "conda-build directory does not exist"
+    exit 1
+fi
+
+# Set staging dir to the working dir, in Windows style if applicable
+if [[ -z "${ARTIFACT_STAGING_DIR}" ]]; then
+    if pwd -W; then
+        ARTIFACT_STAGING_DIR=$(pwd -W)
+    else
+        ARTIFACT_STAGING_DIR=$PWD
+    fi
+fi
+echo "ARTIFACT_STAGING_DIR: $ARTIFACT_STAGING_DIR"
+
+FEEDSTOCK_ROOT=$(cd "$(dirname "$0")/.."; pwd;)
+if [ -z ${FEEDSTOCK_NAME} ]; then
+    export FEEDSTOCK_NAME=$(basename ${FEEDSTOCK_ROOT})
+fi
+
+# Set a unique ID for the artifact(s), specialized for this particular job run
+ARTIFACT_UNIQUE_ID="${CI_RUN_ID}_${CONFIG}"
+if [[ ${#ARTIFACT_UNIQUE_ID} -gt 80 ]]; then
+    ARTIFACT_UNIQUE_ID="${CI_RUN_ID}_${SHORT_CONFIG}"
+fi
+echo "ARTIFACT_UNIQUE_ID: $ARTIFACT_UNIQUE_ID"
+
+# Set a descriptive ID for the archive(s), specialized for this particular job run
+ARCHIVE_UNIQUE_ID="${CI_RUN_ID}_${CONFIG}"
+
+# Make the build artifact zip
+if [[ ! -z "$BLD_ARTIFACT_PREFIX" ]]; then
+    export BLD_ARTIFACT_NAME="${BLD_ARTIFACT_PREFIX}_${ARTIFACT_UNIQUE_ID}"
+    export BLD_ARTIFACT_PATH="${ARTIFACT_STAGING_DIR}/${FEEDSTOCK_NAME}_${BLD_ARTIFACT_PREFIX}_${ARCHIVE_UNIQUE_ID}.zip"
+
+    ( startgroup "Archive conda build directory" ) 2> /dev/null
+
+    # Try 7z and fall back to zip if it fails (for cross-platform use)
+    if ! 7z a "$BLD_ARTIFACT_PATH" "$CONDA_BLD_DIR" '-xr!.git/' '-xr!_*_env*/' '-xr!*_cache/' -bb; then
+        pushd "$CONDA_BLD_DIR"
+        zip -r -y -T "$BLD_ARTIFACT_PATH" . -x '*.git/*' '*_*_env*/*' '*_cache/*'
+        popd
+    fi
+
+    ( endgroup "Archive conda build directory" ) 2> /dev/null
+
+    echo "BLD_ARTIFACT_NAME: $BLD_ARTIFACT_NAME"
+    echo "BLD_ARTIFACT_PATH: $BLD_ARTIFACT_PATH"
+
+    if [[ "$CI" == "azure" ]]; then
+        echo "##vso[task.setVariable variable=BLD_ARTIFACT_NAME]$BLD_ARTIFACT_NAME"
+        echo "##vso[task.setVariable variable=BLD_ARTIFACT_PATH]$BLD_ARTIFACT_PATH"
+    elif [[ "$CI" == "github_actions" ]]; then
+        echo "::set-output name=BLD_ARTIFACT_NAME::$BLD_ARTIFACT_NAME"
+        echo "::set-output name=BLD_ARTIFACT_PATH::$BLD_ARTIFACT_PATH"
+    fi
+fi
+
+# Make the environments artifact zip
+if [[ ! -z "$ENV_ARTIFACT_PREFIX" ]]; then
+    export ENV_ARTIFACT_NAME="${ENV_ARTIFACT_PREFIX}_${ARTIFACT_UNIQUE_ID}"
+    export ENV_ARTIFACT_PATH="${ARTIFACT_STAGING_DIR}/${FEEDSTOCK_NAME}_${ENV_ARTIFACT_PREFIX}_${ARCHIVE_UNIQUE_ID}.zip"
+
+    ( startgroup "Archive conda build environments" ) 2> /dev/null
+
+    # Try 7z and fall back to zip if it fails (for cross-platform use)
+    if ! 7z a "$ENV_ARTIFACT_PATH" -r "$CONDA_BLD_DIR"/'_*_env*/' -bb; then
+        pushd "$CONDA_BLD_DIR"
+        zip -r -y -T "$ENV_ARTIFACT_PATH" . -i '*_*_env*/*'
+        popd
+    fi
+
+    ( endgroup "Archive conda build environments" ) 2> /dev/null
+
+    echo "ENV_ARTIFACT_NAME: $ENV_ARTIFACT_NAME"
+    echo "ENV_ARTIFACT_PATH: $ENV_ARTIFACT_PATH"
+
+    if [[ "$CI" == "azure" ]]; then
+        echo "##vso[task.setVariable variable=ENV_ARTIFACT_NAME]$ENV_ARTIFACT_NAME"
+        echo "##vso[task.setVariable variable=ENV_ARTIFACT_PATH]$ENV_ARTIFACT_PATH"
+    elif [[ "$CI" == "github_actions" ]]; then
+        echo "::set-output name=ENV_ARTIFACT_NAME::$ENV_ARTIFACT_NAME"
+        echo "::set-output name=ENV_ARTIFACT_PATH::$ENV_ARTIFACT_PATH"
+    fi
+fi

--- a/README.md
+++ b/README.md
@@ -170,16 +170,41 @@ conda config --add channels conda-forge
 conda config --set channel_priority strict
 ```
 
-Once the `conda-forge` channel has been enabled, `wxpython` can be installed with:
+Once the `conda-forge` channel has been enabled, `wxpython` can be installed with `conda`:
 
 ```
 conda install wxpython
 ```
 
-It is possible to list all of the versions of `wxpython` available on your platform with:
+or with `mamba`:
+
+```
+mamba install wxpython
+```
+
+It is possible to list all of the versions of `wxpython` available on your platform with `conda`:
 
 ```
 conda search wxpython --channel conda-forge
+```
+
+or with `mamba`:
+
+```
+mamba search wxpython --channel conda-forge
+```
+
+Alternatively, `mamba repoquery` may provide more information:
+
+```
+# Search all versions available on your platform:
+mamba repoquery search wxpython --channel conda-forge
+
+# List packages depending on `wxpython`:
+mamba repoquery whoneeds wxpython --channel conda-forge
+
+# List dependencies of `wxpython`:
+mamba repoquery depends wxpython --channel conda-forge
 ```
 
 
@@ -197,10 +222,12 @@ for each of the installable packages. Such a repository is known as a *feedstock
 A feedstock is made up of a conda recipe (the instructions on what and how to build
 the package) and the necessary configurations for automatic building using freely
 available continuous integration services. Thanks to the awesome service provided by
-[CircleCI](https://circleci.com/), [AppVeyor](https://www.appveyor.com/)
-and [TravisCI](https://travis-ci.com/) it is possible to build and upload installable
-packages to the [conda-forge](https://anaconda.org/conda-forge)
-[Anaconda-Cloud](https://anaconda.org/) channel for Linux, Windows and OSX respectively.
+[Azure](https://azure.microsoft.com/en-us/services/devops/), [GitHub](https://github.com/),
+[CircleCI](https://circleci.com/), [AppVeyor](https://www.appveyor.com/),
+[Drone](https://cloud.drone.io/welcome), and [TravisCI](https://travis-ci.com/)
+it is possible to build and upload installable packages to the
+[conda-forge](https://anaconda.org/conda-forge) [Anaconda-Cloud](https://anaconda.org/)
+channel for Linux, Windows and OSX respectively.
 
 To manage the continuous integration and simplify feedstock maintenance
 [conda-smithy](https://github.com/conda-forge/conda-smithy) has been developed.

--- a/conda-forge.yml
+++ b/conda-forge.yml
@@ -1,3 +1,5 @@
+azure:
+  store_build_artifacts: true
 build_platform:
   osx_arm64: osx_64
 conda_forge_output_validation: true

--- a/recipe/build.sh
+++ b/recipe/build.sh
@@ -47,6 +47,15 @@ fi
 
 env | sort
 
+# Disable code signing during the building stage on osx-arm64
+# The code signing in cctools-port is causing the symbolically linked
+# libraries in $PREFIX/lib to become files which causes issues at runtime
+# https://github.com/conda-forge/wxpython-feedstock/issues/74
+# The environment variable disables the step (defined in cctools-port/cctools/include/stuff/port.h)
+# The libraries are still signed during the packaging stage
+if [[ "$CC" == *"arm64"* ]]; then
+  export NO_CODESIGN=1
+fi
 $PYTHON build.py build_wx install_wx "${PLATFORM_BUILD_FLAGS[@]}" --verbose --no_magic --prefix=$PREFIX --jobs=$CPU_COUNT
 # on macOS --no_magic isn't enough, we need to make build.py use wx-config to find
 # the libraries in ${PREFIX}/lib otherwise they end up being linked in the wxpython

--- a/recipe/meta.yaml
+++ b/recipe/meta.yaml
@@ -18,7 +18,7 @@ source:
     - patches/2354-OS_X-Add-toll-free-bridging-info.patch
 
 build:
-  number: 5
+  number: 6
   osx_is_app: true  # [osx]
   skip: true  # [(win and vc<14) or (osx and py<37)]
   missing_dso_whitelist:   # [osx]

